src/main/java/uk/ac/cam/cl/dtg/isaac/api/PagesFacade.java

```diff
@@ -364,6 +364,26 @@ public final Response getQuestionList(@Context final Request request,
             limit = idsList.size();
         }
 
+        int finalStartIndex = startIndex;
```

Comment:
Using `startIndex` causes an error ("Variable 'startIndex' is accessed from within inner class, needs to be final or effectively final").

Reply:
Oh, I see.

Why not just make a hashmap and add things to it, rather than double-brace initialise it? That would save on making a new class every request, too!